src/main/java/com/synopsys/integration/detect/tool/signaturescanner/BlackDuckSignatureScannerTool.java

```diff
@@ -63,71 +63,90 @@ public BlackDuckSignatureScannerTool(BlackDuckSignatureScannerOptions signatureS
 
     // TODO: Don't accept an Optional as a parameter.
     public SignatureScannerToolResult runScanTool(CodeLocationCreationService codeLocationCreationService, BlackDuckServerConfig blackDuckServerConfig, NameVersion projectNameVersion, Optional<File> dockerTar) throws DetectUserFriendlyException {
-        DetectConfigurationFactory detectConfigurationFactory = detectContext.getBean(DetectConfigurationFactory.class);
         ConnectionFactory connectionFactory = detectContext.getBean(ConnectionFactory.class);
         DirectoryManager directoryManager = detectContext.getBean(DirectoryManager.class);
         CodeLocationNameGenerator codeLocationNameService = detectContext.getBean(CodeLocationNameGenerator.class);
         CodeLocationNameManager codeLocationNameManager = detectContext.getBean(CodeLocationNameManager.class, codeLocationNameService);
```

Comment:
It seems like we're only ever using DetectContext to get objects. While it may be out of scope for this PR, I think ideally we should demand what we need in the method signature or the constructor, rather than getting what we need from detectContext.